### PR TITLE
feat: add order for matchers expression

### PR DESCRIPTION
**Pull request by yyy1000**
Docs for the order of matchers' expression really matters.

Fix: https://github.com/casbin/casbin/issues/1161

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Selflocking plz review

File: docs/SyntaxForModels.mdx [hunk 1]
```diff
@@ -111,6 +111,112 @@ The above matcher is the simplest, it means that the subject, object and action
...
+	}
+
+	request("abu", "/projects/1", "GET")        // really fast because only 2 roles in all policies and at the beginning of the casbin_rule table
+	request("abu", "/projects/2499", "GET")     // fast because only 2 roles in all policies
+	request("jasmine", "/projects/1", "GET")    // really fast at the beginning of the casbin_rule table
+
+	request("jasmine", "/projects/2499", "GET") // slow and fail the only 1st time   <<<< pb here
+	request("jasmine", "/projects/2499", "GET") // fast maybe due to internal cache mechanism
+	
+        // same issue with non-existing roles
```
Comment (Selflocking) [inline docs/SyntaxForModels.mdx:179]: image

please format this line

Comment (yyy1000) [reply docs/SyntaxForModels.mdx]: Thanks, fixed it already.

Comment (Selflocking) [reply docs/SyntaxForModels.mdx]: Please do not mix tabs and spaces

image

You can use vscode (or other editor) to convert all tab to spaces

image

image

Comment (yyy1000) [reply docs/SyntaxForModels.mdx]: Sorry for that. I just copy the code from the author.
Fixed it.

**Review by Selflocking — APPROVED** [review]

**Pull request merged**